apps/client/src/components/Layout/layout.tsx

```diff
@@ -1,15 +1,58 @@
-import { FC } from "react";
-import { LayoutType } from "./type";
-import { Header } from "../Header";
-import { Footer } from "../Footer"
+import { FC, useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { LayoutType } from './type';
+import { Header } from '../Header';
+import { Footer } from '../Footer';
+import { Loader } from '../Loader';
+import RoomService from '../../services/room';
 
 export const Layout: FC<LayoutType> = (props) => {
-    const { children } = props;
-    return (
-        <div>
-            <Header />
-            { children }
-            <Footer />
-        </div>
-    )
-}
\ No newline at end of file
+  const { children } = props;
+  const [loading, setLoading] = useState(false);
+  const location = useLocation();
+  const navigate = useNavigate();
+  const roomKey = new URLSearchParams(location.search).get('roomKey') || '';
+
+  useEffect(() => {
+    const fetchRoomKeyDetails = async () => {
+      try {
+        setLoading(true);
+        const collectRoomData = await RoomService.getRoomById(roomKey);
+        if ('status' in collectRoomData) {
+          if (collectRoomData.status !== 200) {
```

Comment:
we can write wrapper on fetch or axios, which would handle 200 404 all these status and throw somme toaster message